jwst/tweakreg/utils.py

```diff
@@ -84,3 +130,145 @@ def adjust_wcs(wcs, delta_ra=0.0, delta_dec=0.0, delta_roll=0.0,
+        This data model plays two roles: 1) it is the reference WCS which
+        provides a tangent plane in which corrections have been defined, and
+        2) it provides WCS corrections to be applied to ``to_image``'s
+        WCS.
+
+        If the WCS of the ``from_image`` data model does not contain
+        corections, then *both* ``matrix`` *and* ``shift`` arguments *must be
+        supplied*.
+
+    matrix : 2D list, 2D numpy.ndarray, None, optional
+        A 2D matrix part of an affine transformation defined in the tangent
+        plane derived from the ``from_image``'s WCS.
+
+        .. note::
+            When provided, ``shift`` argument *must also be provided* in
+            which case ``matrix`` and ``shift`` arguments override the
+            correction (if present) from the  ``from_file``'s WCS.
+
+    shift : list, numpy.ndarray, None, optional
+        A list of length 2 representing the translational part of an affine
+        transformation (in arcsec) defined in the tangent plane derived
+        from the ``from_image``'s WCS.
+
+        .. note::
+            When provided, ``matrix`` argument *must also be provided* in
+            which case ``matrix`` and ``shift`` arguments override the
+            correction (if present) from the  ``from_file``'s WCS.
+
+    Returns
+    -------
+
+    Upon return, if the ``to_image`` argument is an `ImageModel` it will be
+    modified with an updated ``ImageModel.meta.wcs`` WCS model.
+    If ``to_image`` argument is a file name of an ``ImageModel``, that
+    model will be read in, its WCS will be updated, and the updated model
+    will be written to the same file. BACKUP the file in ``to_image``
+    argument before calling this function.
+
+    """
+    if isinstance(to_image, str):
+        to_file = to_image
+        to_image = ImageModel(to_file)
+    else:
+        to_file = None
+
+    if isinstance(from_image, str):
+        from_image = ImageModel(from_image)
+        refwcs = from_image.meta.wcs
+        ref_wcsinfo = from_image.meta.wcsinfo.instance
+
+    elif isinstance(from_image, WCS):
+        refwcs = from_image
+        ref_wcsinfo = _wcsinfo_from_wcs_transform(refwcs)
+
+    else:  # an ImageModel object
+        refwcs = from_image.meta.wcs
+        ref_wcsinfo = from_image.meta.wcsinfo.instance
+
+    if ((matrix is None and shift is not None) or
+            (matrix is not None and shift is None)):
+        raise ValueError(
+            "Both 'matrix' and 'shift' must be either None or not None."
+        )
+
+    elif matrix is None and shift is None:
+        # use the correction that was applied to the "reference" WCS - the
+        # WCS from the "from_image":
+        if 'v2v3corr' not in refwcs.available_frames:
+            raise ValueError(
+                "The WCS of the 'from_image' data model does not has not been "
+                "previously corrected. A corrected WCS is need in order to "
```

Comment:
minor typo: "need" should be "needed"

Reply:
> One general question: it looks like this will just be available as a function that can be called from the `tweakreg.utils` module, right? It's not a new standalone step in the jwst package.

That is correct. It is a just a tool for post-processing/custom processing. It is not a step. Could be made into a script too, if there is a strong interest.